lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -662,9 +663,9 @@ def create_module_map(native_target)
           end
 
           def module_map_additional_headers
-            return [] unless umbrella_headers_by_dir
+            return [] unless umbrella_headers_dir
 
-            other_paths = umbrella_headers_by_dir[target.module_map_path.dirname] - [target.umbrella_header_path]
+            other_paths = umbrella_headers_dir - [target.umbrella_header_path]
```

Comment:
`umbrella_headers_dir` isn't an array, right? how is this working?

Reply:
turns out it is an array, with 1 element. is there a possibility of there being multiple umbrella headers for a single target?

Reply:
on master it sends an array of umbrella header paths grouped by their directory, so if there is a possibility that one target should include multiple umbrella headers in it's module map path then I should rework this

Reply:
The original grouping logic was implemented here: https://github.com/CocoaPods/CocoaPods/commit/d9e1970644b59295a87a9dc1fcd362890fed7423

Reply:
> turns out it is an array, with 1 element. is there a possibility of there being multiple umbrella headers for a single target?
 
yes, if there are more than two pod targets for the given pod